WordPress/Classes/Utility/Analytics/AnalyticsTrackerAutomatticTracks.swift

```diff
@@ -0,0 +1,241 @@
+            DDLogInfo("WPAnalyticsStat not supported by AnalyticsTrackerAutomatticTracks: \(stat)")
+            return
+        }
+
+        var mergedProperties: [AnyHashable: Any] = event.properties ?? [:]
+        for (key, value) in properties ?? [:] {
+            mergedProperties[key] = value
+        }
+        trackString(event.name, withProperties: mergedProperties)
+    }
+
+    public func trackString(_ event: String) {
+        trackString(event, withProperties: nil)
+    }
+
+    public func trackString(_ event: String, withProperties properties: [AnyHashable: Any]?) {
+        if properties == nil {
+            DDLogInfo("🔵 Tracked: \(event)")
+        } else {
+            let description = (properties ?? [:])
+                .map { (key: "\($0)", value: $1) }
+                .sorted {
+                    $0.key.localizedCaseInsensitiveCompare($1.key) == .orderedAscending
+                }
+                .map { key, value in "\(key): \(value)" }
+                .joined(separator: ", ")
+            DDLogInfo("🔵 Tracked: \(event) <\(description)>")
+        }
+        tracksService.trackEventName(event, withCustomProperties: properties)
+    }
+
+    // MARK: - Session Management
+
+    @objc public func beginSession() {
+        if let currentUserID, !currentUserID.isEmpty {
+            tracksService.switchToAuthenticatedUser(
+                withUsername: currentUserID,
+                userID: nil,
+                wpComToken: try? WPAccount.token(forUsername: currentUserID),
+                skipAliasEventCreation: true
+            )
+        } else {
+            tracksService.switchToAnonymousUser(withAnonymousID: anonymousUserID)
+        }
+        refreshMetadata()
+    }
+
+    @objc public func clearQueuedEvents() {
+        tracksService.clearQueuedEvents()
+    }
+
+    @objc public func refreshMetadata() {
+        let session = getSessionInfo(in: ContextManager.shared.mainContext)
+
+        if let username = session.username, UUID(uuidString: username) != nil {
+            // User has authenticated but we're waiting for account details to sync.
+            // Once details are synced this method will be called again with the actual
+            // username. For now just exit without making changes.
+            return
+        }
+
+        let userProperties = makeUserProperties(with: session)
+        tracksService.userProperties.removeAllObjects()
+        tracksService.userProperties.addEntries(from: userProperties)
+
+        startTracksSession(with: session)
+    }
+
+    private func startTracksSession(with session: SessionInfo) {
+        guard session.isDotcomUser, let username = session.username else {
+            // User is not authenticated, switch to an anonymous mode
+            tracksService.switchToAnonymousUser(withAnonymousID: anonymousUserID)
+            currentUserID = nil
+            return
+        }
+
+        if currentUserID?.isEmpty == true {
```

Comment:
The condition should be `currentUserID?.isEmpty != false` because `currentUserID = nil` should execute the if branch.

Reply:
The previous condition was the following:

```objc
if (self.loggedInID.length == 0) {
```

The logic is that if you don't have any `currentUserID` saved, then call `switchToAuthenticatedUser` with `skipAliasEventCreation: false`.  A `!= nil` check should probably be enough, but I used `isEmpty` just for to be safe.

Reply:
Yeah, I added the comment because the condition check has changed. The previous condition is true if `loggedInID` is nil, but the new condition is false if `currentUserID` is nil.

Reply:
> The condition should be `currentUserID?.isEmpty != false`

It's the same as ` currentUserID?.isEmpty == true`. It's `true` if `currentUserID` is empty.

Reply:
I don't think they are the same. `currentUserID?.isEmpty` has three possible values: `Optional.some(true)`, `.some(false)`, and `Optional.none`. If `currentUserID` is `nil`, `currentUserID?.isEmpty` would be nil and the if condition is false, which is the opssite of the original Objective-C code `if (self.loggedInID.length == 0)`.

Reply:
Oh yeah you are totally right. I don't know who I missed that 🤦 

I didn't want to move the conditions around, so I replaced this condition with the following (not great, but should be clear enough):

```swift
if (currentUserID ?? "").isEmpty
```

In the future, I feel like I'll just stop relying on these optional tricks.